pkg/apis/proxy/proxy.go

```diff
@@ -166,6 +166,7 @@ func handleProxyCommon(p *SprayProxy, c *gin.Context) {
 		copy := c.Copy()
 		newURL := copy.Request.URL
 		newURL.Host = backendURL.Host
+		newURL.Path = backendURL.Path
```

Comment:
I think @enarha's bigger concern is that this is a breaking change (though the right thing to do IMO). We need a feature flag mechanism so current users can disable it.

Reply:
@jduimovich Can you please share if you have specific use case in mind or you are just trying to make sprayproxy more generic. There are different ways to add flexibility here (different endpoint/handler, flags etc) and first I would like to understand what we are trying to achieve here.

Reply:
I was using sprayproxy to take webhooks from Github and forward them to my local machine for PaC webhooks. I use sprayproxy because it has features to `--insecure-skip-tls-verify`  which allows it to talk to my local dev cluster. The machine is firewalled so Github cannot see it directly so coupled with a smee forwarder to sprayproxy, I get webhooks delivered to PaC.

I also take those same web hooks and sent to my argocd instance which has an endpoint `/api/webhook` and sprayproxy removes that  path on the webhook.

So the bug in sprayproxy is that if you send in something like 
`--backend https://rhtap-argocd-server-rhtap.apps-crc.testing/api/webhook`
it will forward that to the `https://rhtap-argocd-server-rhtap.apps-crc.testing`

which is incorrect.

Reply:
If you want to ignore this, please add a warning / error message to sprayproxy that prints an error and exits with an error code and reports that  providing a path on your --backend endpoint is not supported.